### PR TITLE
Templates GET API & Basic QuantumLab Agent

### PR DESCRIPTION
**Pull request by QuanchiChen**
1. Update the templates GET API endpoint to return all the authorised templates.
2. Develop QuantumLab Agent.
3. Develop the agent POST API endpoint.

### CODE REVIEW

**Review by StevenDB995 — APPROVED** [review]

**Review by lois-ekd81194 — APPROVED** [review]

File: agent/config.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+metadata:
```
Comment (VortexMashiro) [inline agent/config.yaml:1]: please add example config to readme, not in repository

File: agent/go.mod [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module github.com/Project-Quantum-Workspace/QuantumLab/agent
+
+go 1.19
```
Comment (VortexMashiro) [inline agent/go.mod:3]: Use same version with Quantumlab main pls, which is 1.20

File: agent/initialisation.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
```
Comment (VortexMashiro) [inline agent/initialisation.sh:1]: same here, please add example files to readme

**Review by VortexMashiro — CHANGES_REQUESTED** [review]
Please do some small changes refer to comments :)

**Review by VortexMashiro — APPROVED** [review]

**Pull request merged**